lib/common_fw/common_fw.cpp

```diff
@@ -36,36 +41,34 @@ bool write_to_SD(std::string message, const char * file_name) {
  * Constructs packet in the following format:
  * {<sensor_ID>,<data1>,<data2>, ...,<dataN>|checksum}
  */
-String make_packet(int id, bool error) {
-  String packet_content = (String)id;
-  packet_content += ",";
+uint8_t make_packet(uint8_t id, bool error) {
+  int i = 0;
+  packet[i++] = (PACKET_START >> 8) & 0xFF;
+  packet[i++] = PACKET_START & 0xFF;
+
+  packet[i++] = id;
   if (!error) {
-    for (int i=0; i<7; i++) {
-      float reading = farrbconvert.sensorReadings[i];
-      if (reading != -1) {
-        packet_content += (String)reading;
-        packet_content += ",";
+    for (int j=0; j<7; j++) {
```

Comment:
So this means we can have up to 6 floats in the packet?

Reply:
7

Reply:
Yeet